src/connectionconfig/connectionDialogWebViewController.ts

```diff
@@ -314,38 +293,129 @@ export class ConnectionDialogWebViewController extends ReactWebViewPanelControll
 						validationMessage: ''
 					};
 				}
-			},
-			{
-				propertyName: 'database',
-				label: 'Database',
-				required: false,
-				type: FormComponentType.Input,
-			},
-			{
-				propertyName: 'trustServerCertificate',
-				label: 'Trust Server Certificate',
-				required: false,
-				type: FormComponentType.Checkbox,
-			},
+			}
+		});
+
+		components.set('profileName', {
+			option: undefined,
+			component: {
+			propertyName: 'profileName',
+			label: 'Profile Name',
+			required: false,
+			type: FormComponentType.Input,
+			}
+		});
+
+		// add missing validation functions for generated components
+		components.get('server')!.component.validate = (value: string) => {
+			if (this.state.selectedFormTab === FormTabType.Parameters && !value) {
+				return {
+					isValid: false,
+					validationMessage: 'Server is required'
+				};
+			}
+			return {
+				isValid: true,
+				validationMessage: ''
+			};
+		};
+
+		components.get('user')!.component.validate = (value: string) => {
+			if (this.state.connectionProfile.authenticationType === AuthenticationType.SqlLogin && !value) {
+				return {
+					isValid: false,
+					validationMessage: 'User name is required'
+				};
+			}
+			return {
+				isValid: true,
+				validationMessage: ''
+			};
+		};
+
+		components.get('tenantId')!.component.validate = (value: string) => {
+			if (this.state.connectionProfile.authenticationType === AuthenticationType.AzureMFA && !value) {
+				return {
+					isValid: false,
+					validationMessage: 'Tenant ID is required'
+				};
+			}
+			return {
+				isValid: true,
+				validationMessage: ''
+			};
+		};
+	}
+
+	private _mainOptionNames = new Set<string>([
+		'server',
+		'authenticationType',
+		'user',
+		'password',
+		'savePassword',
+		'accountId',
+		'tenantId',
+		'database',
+		'trustServerCertificate',
+		'encrypt',
+		'profileName'
+	]);
+
+	private async generateConnectionFormComponents(): Promise<{
+		mainComponents: FormComponent[],
+		advancedComponents: {[category: string]: FormComponent[]}
+	}> {
+		// get list of connection options from Tools Service
+		const result: CapabilitiesResult = await this._mainController.connectionManager.client.sendRequest(GetCapabilitiesRequest.type, {});
+		const connectionOptions: ConnectionOption[] = result.capabilities.connectionProvider.options;
+
+		// convert connection options to form components
+		const allConnectionFormComponents = new Map<string, {option: ConnectionOption, component: FormComponent}>();
+
+		for (const option of connectionOptions) {
+			allConnectionFormComponents.set(option.name, {option, component: this.convertToFormComponent(option)});
+		}
+
+		await this.completeFormComponents(allConnectionFormComponents);
+
+		// organize the main components and advanced components
+		const mainComponents: FormComponent[] = [];
```

Comment:
add comment saying that advanced components are grouped, while main components aren't.